packages/dataviews/src/view-grid.tsx

```diff
@@ -40,7 +46,6 @@ interface GridItemProps< Item extends AnyItem > {
 
 function GridItem< Item extends AnyItem >( {
 	selection,
-	data,
```

Comment:
Should this get removed from `GridItemProps` type.

Reply:
Hm, why did TS not complain?

Reply:
It looks like `AnyItem` is `Record<string, any>`, so any string key is associated with a value of type `any`. `Item` extends `AnyItem`, so `Item` must also have `string` -> `any` mapping. This weakens the type system, it would be preferable to have a better type to extend or remove the `AnyItem` type somehow.

Reply:
@sirreal You're right, `AnyItem` is currently hiding several type errors, because it pollutes the types with `any`, and also it turns out to be redundant, can be removed completely. We discussed it already in https://github.com/WordPress/gutenberg/pull/62647#discussion_r1648576427 and here's my draft PR (on top of #62647) that removes `AnyItem` and fixes the three type error that the removal has revealed: #62856.